src/parser/ast/enums.rs

```diff
@@ -0,0 +1,62 @@
+use crate::lexer::token::Span;
+use crate::parser::ast::attributes::AttributeGroup;
+use crate::parser::ast::classish::ClassishConstant;
+use crate::parser::ast::functions::Method;
+use crate::parser::ast::identifiers::Identifier;
+use crate::parser::ast::Expression;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UnitEnumCase {
+    pub start: Span,
+    pub end: Span,
+    pub name: Identifier,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum UnitEnumMember {
+    Case(UnitEnumCase),
+    Method(Method),
+    Constant(ClassishConstant),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct UnitEnum {
+    pub start: Span,
+    pub end: Span,
+    pub name: Identifier,
+    pub attributes: Vec<AttributeGroup>,
+    pub implements: Vec<Identifier>,
+    pub members: Vec<UnitEnumMember>,
+}
```

Comment:
this is how i hope to define other types as well, `Method` and `Property`aren't statements IMO, they are a member of a type, so i they shouldn't live there.